### PR TITLE
[2902] Eliminate `ChatDomain::deleteChannel`

### LINKED ISSUE #2897: Eliminate ChatDomain::deleteMessage
### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Side effects of edit message are moved to OfflinePlugin
- [ ] Calls replaced by ChatClient::deleteMessage
- [ ] Method is deprecated

### LINKED ISSUE #2902: Eliminate ChatDomain::deleteChannel
### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Side effects of deleteChannel are moved to OfflinePlugin
- [ ] Calls replaced by ChatClient::deleteChannel
- [ ] Method is deprecated

### PR DESCRIPTION
**Pull request by amitkma**
Closes #2902 

### 🎯 Goal

In order to merge `ChatDomain` into LLC, we need to move all code from ChatDomain into LLC and `OfflinePlugin`.

### 🛠 Implementation details
- A new `ChatClient::deleteChannel` extension function is added.
- `DeleteChannel` usecase is removed.
- `ChatDomain::deleteChannel` is deprecated.

### 🧪 Testing

1. Run any sample app.
2. Try to Delete any channel for which you have permission to delete.
2. Make sure the channel is deleted as expected. 

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/ChatInfoViewModel.kt [hunk 1]
```diff
@@ -134,7 +135,7 @@ class ChatInfoViewModel(
     private fun deleteChannel() {
```
Comment (filbabic) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/ChatInfoViewModel.kt:135]: Some docs around here would be great!

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt [hunk 2]
```diff
@@ -249,7 +250,7 @@ public class ChannelListViewModel(
     }
 
     public fun deleteChannel(channel: Channel) {
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt:252]: And here!

**Review by filbabic — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

File: CHANGELOG.md [hunk 3]
```diff
@@ -41,10 +41,13 @@ Consider migrating to `stream-chat-android-ui-components` or `stream-chat-androi
 ### ⬆️ Improved
 
 ### ✅ Added
+- Added new extension function `ChatClient::deleteChannel`. [#3007](https://github.com/GetStream/stream-chat-android/pull/3007)
 
 ### ⚠️ Changed
+- Deprecated `ChatDomain::deleteChannel` in favour of `ChatClient::deleteChannel`. [#3007](https://github.com/GetStream/stream-chat-android/pull/3007)
 
 ### ❌ Removed
+- `DeleteChannel` usecase is now removed. [#3007](https://github.com/GetStream/stream-chat-android/pull/3007)
```
Comment (adasiewiczr) [inline CHANGELOG.md:50]: `DeleteChannel` use case was internal so we don't need to mention that here

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt [hunk 4]
```diff
@@ -303,3 +305,15 @@ internal fun ChatClient.shuffleGiphy(message: Message): Call<Message> {
...
+
+/**
+ * Deletes the channel with the specified id.
+ *
+ * @param cid The full channel id i. e. messaging:123.
+ *
+ * @return Executable async [Call] deleting a channel.
+ */
+@CheckResult
+public fun ChatClient.deleteChannel(cid: String): Call<Unit> = CoroutineCall(domainImpl().scope) {
```
Comment (adasiewiczr) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt:317]: Why do we need this extension? I think we should suggest using `ChatClient.instance().channel(cid).delete()` or `ChatClient.instance().delete(channelType, channelId)` and skip it. WDYT?

File: CHANGELOG.md (hunk 3 above)
Comment (amitkma) [reply CHANGELOG.md]: Done

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt (hunk 4 above)
Comment (amitkma) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt]: Done

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomain.kt [hunk 5]
```diff
@@ -561,6 +561,14 @@ public sealed interface ChatDomain {
      * @return Executable async [Call] deleting a channel.
      */
     @CheckResult
+    @Deprecated(
+        message = "deleteChannel is deprecated. Use function ChatClient::deleteChannel instead",
+        replaceWith = ReplaceWith(
+            expression = "ChatClient.instance().deleteChannel(cid)",
```
Comment (adasiewiczr) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomain.kt:567]: We should also update those expressions, right?

**Review by adasiewiczr — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

**Pull request merged**